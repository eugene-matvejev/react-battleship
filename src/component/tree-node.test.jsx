import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TreeNode from './tree-node';

configure({ adapter: new Adapter() });

describe('<TreeNode/>', () => {
    const props = {
        text: '{{text}}',
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<TreeNode {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['className', '{{className}}'],
                ['data-cy', '{{data-cy}}'],
                [
                    'chunks',
                    [
                        { v: 'a', isMatch: true },
                        { v: 'b', isMatch: false },
                        { v: 'c' },
                    ],
                ],
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<TreeNode {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });

            describe('prop combinations', () => {
                [
                    [
                        'collapsed with nodes',
                        [
                            ['isExpanded', false],
                            ['nodes', [{ text: '{{node-text}}' }]],
                        ],
                    ],
                    [
                        'expanded with NO nodes',
                        [
                            ['isExpanded', true],
                            ['nodes', []],
                        ],
                    ],
                    [
                        'expanded with "hidden" nodes',
                        [
                            ['isExpanded', true],
                            ['nodes', [{ text: '{{node-text}}' }]],
                        ],
                    ],
                    [
                        'expanded with "visible" nodes',
                        [
                            ['isExpanded', true],
                            ['nodes', [{ text: '{{node-text}}', isVisible: true }]],
                        ],
                    ],
                ].forEach(([desc, v]) => {
                    it(`::${v.map(([p]) => p).join(', ::')} - ${desc}`, () => {
                        const c = shallow(<TreeNode
                            {...props}
                            {...v.reduce((acc, [prop, v]) => (acc[prop] = v, acc), {})}
                        />);

                        expect(c).toMatchSnapshot();
                    });
                });
            });
        });
    });
});
