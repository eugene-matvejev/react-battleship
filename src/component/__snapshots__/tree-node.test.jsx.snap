// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreeNode/> render with default/required props 1`] = `
<div
  className="tree-node "
  data-cy=""
  data-node={0}
>
  <span>
    {{text}}
  </span>
</div>
`;

exports[`<TreeNode/> render with optional props [::chunks] as "[object Object],[object Object],[object Object]" 1`] = `
<div
  className="tree-node "
  data-cy=""
  data-node={0}
>
  <span
    className="tree-node--pattern-match"
    key="0"
  >
    a
  </span>
  <span
    className={null}
    key="1"
  >
    b
  </span>
  <span
    className={null}
    key="2"
  >
    c
  </span>
</div>
`;

exports[`<TreeNode/> render with optional props [::className] as "{{className}}" 1`] = `
<div
  className="tree-node {{className}}"
  data-cy=""
  data-node={0}
>
  <span>
    {{text}}
  </span>
</div>
`;

exports[`<TreeNode/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<div
  className="tree-node "
  data-cy="{{data-cy}}"
  data-node={0}
>
  <span>
    {{text}}
  </span>
</div>
`;

exports[`<TreeNode/> render with optional props prop combinations ::isExpanded, ::nodes - collapsed with nodes 1`] = `
<div
  className="tree-node--with-children "
  data-cy=""
>
  <span>
    {{text}}
  </span>
</div>
`;

exports[`<TreeNode/> render with optional props prop combinations ::isExpanded, ::nodes - expanded with "hidden" nodes 1`] = `
<div
  className="tree-node--with-children-expanded "
  data-cy=""
>
  <span>
    {{text}}
  </span>
</div>
`;

exports[`<TreeNode/> render with optional props prop combinations ::isExpanded, ::nodes - expanded with "visible" nodes 1`] = `
<div
  className="tree-node--with-children-expanded "
  data-cy=""
>
  <span>
    {{text}}
  </span>
  <TreeNode
    className=""
    data-cy="-0"
    data-node="undefined-0"
    isVisible={true}
    key="0"
    text="{{node-text}}"
  />
</div>
`;

exports[`<TreeNode/> render with optional props prop combinations ::isExpanded, ::nodes - expanded with NO nodes 1`] = `
<div
  className="tree-node--with-children-expanded "
  data-cy=""
>
  <span>
    {{text}}
  </span>
</div>
`;
